File: gb-angular-frontend/src/app/interfaces/review.ts
import {ReviewOverallScore} from './review-overall-score';

export interface Review {
  review_id: number;
  review_subject: string;
  review_content: string;
  score: number;
  poster_id: number;
  poster_username: string;
  product_id: number;
  product_title: string;
  package_id: number;
  package_name: string;
  seller_id: number;
  seller_username: string;
  post_date: string;
  status: string;
  poster_icon: string;
  review_overall_score?: ReviewOverallScore;
}

File: gb-spring-backend/src/main/java/com/egroupbuy/model/Review.java
package com.egroupbuy.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Timestamp;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Review {
    private int review_id;
    private String review_subject;
    private String review_content;
    private int score;
    private int poster_id;
    private String poster_username;
    private int product_id;
    private String product_title;
    private int package_id;
    private String package_name;
    private int seller_id;
    private String seller_username;
    private Timestamp post_date;
    private String status;
    private String poster_icon;
    private ReviewOverallScore review_overall_score;
}
